src/redfish_collector/routers/prometheus.py

```
@@ -2,107 +2,97 @@
-                    #     inventory.remove(server)
-                if existed is False:
-                    inventory.append(block)
-                with open('%sinventory.yml' %REDFISH_DATA, 'w') as f:
-                    yaml.dump(inventory, f, default_flow_style=False)
+                with open(inventoryFile, 'w') as f:
+                    yaml.dump([block], f, default_flow_style=False)
         except Exception as err:
-            logging.error("Generate instance failed: %s" %err)
-            return False
-
+            logging.exception("Generate instance failed: %s" % err)
```

Ruff (0.14.7)

Redundant exception object included in `logging.exception` call

(TRY401)